ci/infra/testrunner/skuba/skuba.py

```diff
@@ -154,6 +156,10 @@ def node_remove(self, role="worker", nr=0):
     def node_upgrade(self, action, role, nr, ignore_errors=False):
         self._verify_bootstrap_dependency()
 
+        self.cluster_init("", "", "/tmp")
+        cmd = "rsync -a /tmp/test-cluster/addons/ addons/"
```

Comment:
I don't like this hardcoded path for several reasons. First, In the CI, for example it is unsafe to use `/tmp`. I would be safer to create a `tmp` directory in the workspace directory.  

Also, instead of hardcoding `test-cluster` you could better use the configuration parameter `self.config.skuba.cluster` because the `cluster_init` function will use it.